frontend/src/metabase/reference/reference.js

```diff
@@ -31,42 +49,499 @@ export const fetchGuide = createThunkAction(FETCH_GUIDE, (reload = false) => {
+
+    fetchDataWrapper(
+        props, 
+        async (mID) => { 
+                await Promise.all(
+                    [props.fetchMetricTable(mID),
+                     props.fetchMetrics(), 
+                     props.fetchQuestions()]
+                )}
+        )(metricID)
+}
+export const wrappedFetchMetricRevisions = async (props, metricID) => {
+
+    fetchDataWrapper(
+        props, 
+        async (mID) => { 
+                await Promise.all(
+                    [props.fetchMetricRevisions(mID),
+                     props.fetchMetrics()]
+                )}
+        )(metricID)
+}
+
+// export const wrappedFetchDatabaseMetadataAndQuestion = async (props, databaseID) => {
+//         clearError();
+//         startLoading();
+//         try {
+//             await Promise.all(
+//                     [props.fetchDatabaseMetadata(databaseID),
+//                      props.fetchQuestions()]
+//                 )
+//         }
+//         catch(error) {
+//             console.error(error);
+//             setError(error);
+//         }
+
+//         endLoading();
+// }
+
+export const wrappedFetchDatabases = (props) => {
+    fetchDataWrapper(props, props.fetchDatabases)({})
+}
+export const wrappedFetchMetrics = (props) => {
+    fetchDataWrapper(props, props.fetchMetrics)({})
+}
+
+export const wrappedFetchSegments = (props) => {
+    fetchDataWrapper(props, props.fetchSegments)({})
+}
+
+
+export const wrappedFetchSegmentDetail = (props, segmentID) => {
+    fetchDataWrapper(props, props.fetchSegmentTable)(segmentID)
+}
+
+export const wrappedFetchSegmentQuestions = async (props, segmentID) => {
+
+    fetchDataWrapper(
+        props, 
+        async (sID) => { 
+                await props.fetchSegments(sID);
+                await Promise.all(
+                    [props.fetchSegmentTable(sID),
+                     props.fetchQuestions()]
+                )}
+        )(segmentID)
+}
+export const wrappedFetchSegmentRevisions = async (props, segmentID) => {
+
+    fetchDataWrapper(
+        props, 
+        async (sID) => { 
+                await props.fetchSegments(sID);
+                await Promise.all(
+                    [props.fetchSegmentRevisions(sID),
+                     props.fetchSegmentTable(sID)]
+                )}
+        )(segmentID)
+}
+export const wrappedFetchSegmentFields = async (props, segmentID) => {
+
+    fetchDataWrapper(
+        props, 
+        async (sID) => { 
+                await props.fetchSegments(sID);
+                await Promise.all(
+                    [props.fetchSegmentFields(sID),
+                     props.fetchSegmentTable(sID)]
+                )}
+        )(segmentID)
+}
+
+// This is called when a component gets a new set of props.
+// I *think* this is un-necessary in all cases as we're using multiple 
+// components where the old code re-used the same component
+export const clearState = props => {
```

Comment:
It's slightly confusing that this function isn't actually an action creator but it's in the "duck" file along with other actions, and being called like `actions.clearState(newProps)`. It also feels a little weird to have any mention of "props" in Redux code.

We could make it an actual thunk action creator that dispatches each of the other actions. Or move it (and `resetForm` and any other similar fns) to another file. Or just leave it as is and document this minor weirdness.